tools/system_libs.py

```diff
@@ -261,32 +261,38 @@ def create_compiler_rt(libname):
     shared.Building.emar('cr', in_temp(libname), o_s)
     return in_temp(libname)
 
-  def create_dlmalloc(out_name, clflags):
+  def dlmalloc_name():
+    ret = 'dlmalloc'
+    if shared.Settings.USE_PTHREADS:
+      ret += '_threadsafe'
+    if shared.Settings.EMSCRIPTEN_TRACING:
+      ret += '_tracing'
+    if shared.Settings.SPLIT_MEMORY:
+      ret += '_split'
+    if shared.Settings.DEBUG_LEVEL:
+      ret += '_debug'
```

Comment:
Looks good, btw, is this the first instance where passing `-g1` or bigger affects codegen? I.e. earlier `-gX` have only affected whitespace, function names and minification?